tests/macsec/test_macsec_mka_multiple_devices.py

```diff
@@ -0,0 +1,340 @@
+ 
+    ####################
+    # MKA
+    ####################
+    secy1_key_gen_proto, secy2_key_gen_proto = secy1.key_generation_protocol, secy2.key_generation_protocol
+    secy3_key_gen_proto, secy4_key_gen_proto = secy3.key_generation_protocol, secy4.key_generation_protocol
+
+    secy1_key_gen_proto.choice = secy2_key_gen_proto.choice = "mka"
+    secy3_key_gen_proto.choice = secy4_key_gen_proto.choice = "mka"
+
+    kay1, kay2 = secy1_key_gen_proto.mka, secy2_key_gen_proto.mka
+    kay3, kay4 = secy3_key_gen_proto.mka, secy4_key_gen_proto.mka
+
+    kay1.name, kay2.name = "mka1", "mka2"
+    kay3.name, kay4.name = "mka3", "mka4"
+
+    # Basic properties
+    kay1.basic.key_derivation_function = kay2.basic.key_derivation_function = "aes_cmac_128"
+    kay3.basic.key_derivation_function = kay4.basic.key_derivation_function = "aes_cmac_128"
+
+    # Key source: PSK
+    kay1_key_src, kay2_key_src = kay1.basic.key_source, kay2.basic.key_source
+    kay3_key_src, kay4_key_src = kay3.basic.key_source, kay4.basic.key_source
+
+    kay1_key_src.choice = kay2_key_src.choice = "psk"
+    kay3_key_src.choice = kay4_key_src.choice = "psk"
+
+    kay1_psk_chain, kay2_psk_chain = kay1_key_src.psks, kay2_key_src.psks
+    kay3_psk_chain, kay4_psk_chain = kay3_key_src.psks, kay4_key_src.psks
+
+    # PSK 1
+    kay1_psk1, kay2_psk1 = kay1_psk_chain.add(), kay2_psk_chain.add()
+    kay3_psk1, kay4_psk1 = kay3_psk_chain.add(), kay4_psk_chain.add()
+
+    kay1_psk1.cak_name = kay2_psk1.cak_name = kay3_psk1.cak_name = kay4_psk1.cak_name = "0xF123456789ABCDEF0123456789ABCDEFF123456789ABCDEF0123456789ABCD01"
+    kay1_psk1.cak_value = kay2_psk1.cak_value = kay3_psk1.cak_value = kay4_psk1.cak_value = "0xF123456789ABCDEF0123456789ABCD01"
+
+    kay1_psk1.start_offset_time.hh = kay2_psk1.start_offset_time.hh = kay3_psk1.start_offset_time.hh = kay4_psk1.start_offset_time.hh = 0
+    kay1_psk1.start_offset_time.mm = kay2_psk1.start_offset_time.mm = kay3_psk1.start_offset_time.mm = kay4_psk1.start_offset_time.mm = 0
+
+    kay1_psk1.end_offset_time.hh = kay2_psk1.end_offset_time.hh = kay3_psk1.end_offset_time.hh = kay4_psk1.end_offset_time.hh = 0
+    kay1_psk1.end_offset_time.hh = kay2_psk1.end_offset_time.hh = kay3_psk1.end_offset_time.hh = kay4_psk1.end_offset_time.hh = 0
+
+    # Rekey mode
+    kay1_rekey_mode, kay2_rekey_mode, kay3_rekey_mode, kay4_rekey_mode = kay1.basic.rekey_mode, kay2.basic.rekey_mode, kay3.basic.rekey_mode, kay4.basic.rekey_mode
+    kay1_rekey_mode.choice = kay2_rekey_mode.choice = kay3_rekey_mode.choice = kay4_rekey_mode.choice = "dont_rekey"
+
+    # Remaining basic properties autofilled
+
+    # Key server
+    kay1_key_server, kay2_key_server, kay3_key_server, kay4_key_server = kay1.key_server, kay2.key_server, kay3.key_server, kay4.key_server
+    kay1_key_server.cipher_suite = kay2_key_server.cipher_suite = kay3_key_server.cipher_suite = kay4_key_server.cipher_suite = "gcm_aes_128"
```

Comment:
Please add validation something like 
https://github.com/open-traffic-generator/snappi-ixnetwork/blob/main/tests/bgp/test_bgp_attributes.py#L134

Please search with `api._ixnetwork` within test. You will get more such examples.